python/ray/util/spark/utils.py

```diff
@@ -369,6 +369,7 @@ def _get_num_physical_gpus():
             check=True,
             text=True,
             capture_output=True,
+            timeout=60,
```

Comment:
Actually Ray now has vendored pynvml library and you can use that to get the number of physical gpus. Check `nvidia_gpu.py`: 

```
    @staticmethod
    def get_current_node_num_accelerators() -> int:
        import ray._private.thirdparty.pynvml as pynvml

        try:
            pynvml.nvmlInit()
        except pynvml.NVMLError:
            return 0  # pynvml init failed
        device_count = pynvml.nvmlDeviceGetCount()
        pynvml.nvmlShutdown()
        return device_count
```

You can actually call this method directly.